examples/counter/store/configureStore.js

```diff
@@ -1,11 +1,21 @@
-import { createStore, applyMiddleware } from 'redux';
+import { createStore, applyMiddleware, compose } from 'redux';
 import thunk from 'redux-thunk';
-import rootReducer from '../reducers';
+import reducer from '../reducers';
+import switchReducer from './switchReducer';
 
-const createStoreWithMiddleware = applyMiddleware(
-  thunk
-)(createStore);
+function onHotReload(useReducer) {
```

Comment:
Not sure I understand what you're proposing.

The problem is Webpack needs `module.hot.accept(name` call to be verbatim and statically analyzable (also true for `require` call inside).

Also, `switchReducer` argument isn't specific to hot reloading. It's just a function that invokes a callback whenever there's a new reducer available. This can be a result of hot reloading, code splitting, or even both.